integration/tests/cook/test_basic.py

```diff
@@ -551,6 +551,34 @@ def test_change_retries(self):
         job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
         self.assertEqual('success', job['state'], 'Job details: %s' % (json.dumps(job, sort_keys=True)))
 
+    def test_change_failed_retries(self):
+        job_specs = util.minimal_jobs(2, max_retries=1, command='sleep 5')
+        try:
+            jobs, resp = util.submit_jobs(self.cook_url, job_specs)
+            self.assertEqual(resp.status_code, 201)
+            # wait for first job to start running, and kill it
+            job_uuid = jobs[0]
+            util.wait_for_job(self.cook_url, job_uuid, 'running')
+            util.kill_jobs(self.cook_url, [job_uuid])
+            job = util.load_job(self.cook_url, job_uuid)
+            self.assertEqual('failed', job['state'])
```

Comment:
I think we have a race between the kill and the job completing. If the job completes before the kill, the job state will not be "failed". Maybe we should make it sleep longer than 5 seconds?

Reply:
Bumped up to 10 seconds.

Reply:
Done. (We're killing the jobs now instead of waiting for completion, so the longer sleep won't hurt us.)